vendor/src/github.com/docker/engine-api/types/auth.go

```diff
@@ -5,7 +5,6 @@ type AuthConfig struct {
 	Username      string `json:"username,omitempty"`
 	Password      string `json:"password,omitempty"`
 	Auth          string `json:"auth,omitempty"`
-	Email         string `json:"email"`
```

Comment:
With further testing, if you login with change, then you can still use the config file to login to other servers and push from an older client, just an attempt to login again will fail. With the omitempty flag added, the email will not be added when logging in with this change so downgrade login will still fail, but if you previously logged in with older client, logging in with this change will not strip the email. I don't think it will have a hugely negative impact to completely remove but still think omitempty is probably the correct solution. WDYT?